bin/compute-classpath.sh

```diff
@@ -121,10 +124,15 @@ datanucleus_jars="$(find "$datanucleus_dir" 2>/dev/null | grep "datanucleus-.*\\
 datanucleus_jars="$(echo "$datanucleus_jars" | tr "\n" : | sed s/:$//g)"
 
 if [ -n "$datanucleus_jars" ]; then
-  hive_files=$("$JAR_CMD" -tf "$ASSEMBLY_JAR" org/apache/hadoop/hive/ql/exec 2>/dev/null)
-  if [ -n "$hive_files" ]; then
-    echo "Spark assembly has been built with Hive, including Datanucleus jars on classpath" 1>&2
-    CLASSPATH="$CLASSPATH:$datanucleus_jars"
+  if [ $(command -v "$JAR_CMD") ] ; then
+    hive_files=$("$JAR_CMD" -tf "$ASSEMBLY_JAR" org/apache/hadoop/hive/ql/exec 2>/dev/null)
+    if [ -n "$hive_files" ]; then
+      echo "Spark assembly has been built with Hive, including Datanucleus jars on classpath" 1>&2
+      CLASSPATH="$CLASSPATH:$datanucleus_jars"
+    fi
+  else
+    echo "No jar command available; unable to check if assembly was built with Hive."
```

Comment:
Yeah, this upgrades silence to a warning; upgrading it to an error however means that a JDK is suddenly required to run the thrift server et al, which does not seem like something that should be required (production usually disallows JDKs). However, yes, my question is: why not just always add these JARs if there's doubt, and they're present? They could be added in the `else` block, or, why bother inspecting what is built into the assembly anyway?